src/edu/csus/ecs/pc2/ui/SelectJudgementPaneNew.java

```diff
@@ -1519,6 +1528,17 @@ protected void viewOutputsAndData() {
     private TestResultsFrame getTestResultsFrame() {
         if (testResultsFrame == null) {
             testResultsFrame = new TestResultsFrame();
+            // Add action listener whether Test results pane button is clicked
+            testResultsFrame.addActionListenerToExecuteAll(new ActionListener() {
+                @Override
+                public void actionPerformed(ActionEvent e) {
+                    new Thread(new Runnable() {
+                        public void run() {
+                            executeRun(true);
+                        }
+                    }).start();
+                }
+            });
```

Comment:
This block of code (adding an `actionListener()` to the `ExecuteAll` button) is a bit awkward; normally, code like this belongs in the singleton accessor/constructor (in this case, in `TestResultsPane.getExecuteAllButton()`, right after the `executeAllButton = new JButton()` statement).  As presently coded, this block makes calls through a series of methods in different classes (`TestResultsFrame.addActionListenerToExecuteAll()` --> `TestResultsPane.addActionListenerToExecuteAll()` --> `TestResultsPane.getExecuteAllButton()`).  If changes are made later to methods in that sequence, it can result in an error in the calling class (`SelectJudgementPaneNew`).   In other words, there is a lot of tight coupling between these classes, which in general is not good.

I do recognize that moving this block of code into the singleton accessor/constructor would raise the problem of how the ActionListener code is going to be able to execute the statement `executeRun(true);` since the `execute()` method is in the `SelectJudgementPaneNew` class, not in the `TestResultsPane` class where the `ExecuteAll` button is defined.  I suspect that moving the code to the `ExecuteAll` constructor would require adding code to allow it to reference back to the `SelectJudgementPaneNew.execute()` method, so I'm not necessarily requesting a change.  It just seems awkward to have this long sequence of `addActionListenerToExecuteAll()` method calls.

Reply:
I kind of figured out a way to fix this however one small issue that arises is that the Execute All button now comes up in the end after the Close button. So far, I'm unable to figure out how to change the ordering of the buttons. I'm still looking into it.

![image](https://github.com/pc2ccs/pc2v9/assets/24360328/fcbe960f-3eed-4d78-9e7c-9745fa905199)

Reply:
One workaround for it can be that `TestResultsPane` object should have an object of `SelectJudgementPaneNew` as a member variable. When an object of `TestResultsPane` is created inside `SelectJudgementPaneNew` we can set the `TestResultsPane` 's member with `this` and use the `executeRun` function from there.

Reply:
Yes, this is one approach to solve the problem -- instead of having the parent (composing) class (`SelectJudgementPaneNew`) have references downward to the composed classes (like `TestResultsFrame` and `TestResultsPane`), support the ability of the composed class (`TestResultsPane`) to have a reference upward to its composing class.  This would be slightly cleaner, although it still represents tight coupling between the classes.

In any case, if you take that approach I don't think method `SelectJudgementPaneNew.executeRun()` should be _public_; it should be _protected_ -- which will allow classes in the same package to access its methods while prohibiting general public access.  This will work since both `SelectJudgementPaneNew` and `TestResultsPane` are in the same package (`edu.csus.ecs.pc2.ui`).